### PR TITLE
[AF-2310]: Importing samples and projects are now subdirectory clones

### PR DESCRIPTION
**Pull request by adrielparedes**
Two changes are included in this PR and the biggest changes are in here.

## Deprecated project

kie-wb-common-screen-examples-client was deprecated some time ago but never removed. Now it was removed from all kiegroup.

## Copy to Clone

Samples projects were copied file by file and that generated a lot of overhead when importing projects. It was replaced by clone subdirectory operation that improves speed and reliability of the imported project.

## Related to: 
* https://github.com/kiegroup/appformer/pull/838
* https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1092
* https://github.com/kiegroup/kie-wb-common/pull/2991
* https://github.com/kiegroup/jbpm-wb/pull/1401
* https://github.com/kiegroup/drools-wb/pull/1256
* https://github.com/kiegroup/kie-wb-distributions/pull/986
* https://github.com/kiegroup/optaplanner-wb/pull/349

### CODE REVIEW

**Comment by paulovmr** [comment]
Part of an ensemble:
 * https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1092
 * https://github.com/kiegroup/kie-wb-common/pull/2991

File: kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/pom.xml [hunk 1]
```diff
@@ -177,6 +177,12 @@
       <artifactId>uberfire-testing-utils</artifactId>
       <scope>test</scope>
     </dependency>
+
+    <dependency>
```
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/pom.xml]: Where was this used?

File: kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/src/main/java/org/kie/workbench/common/screens/examples/backend/validation/NameAlreadyExistValidator.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    @Inject
+    public NameAlreadyExistValidator(ProjectImportService projectImportService) {
+
+        this.projectImportService = projectImportService;
+    }
+
+    @Override
+    protected Optional<ExampleProjectError> getError(OrganizationalUnit ou, ImportProject importProject) {
+        if (this.projectImportService.exist(ou, importProject)) {
+            String errorPattern = "Already exist project <{1}> in space <{0}> ";
```
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/src/main/java/org/kie/workbench/common/screens/examples/backend/validation/NameAlreadyExistValidator.java]: Minor fix: `"Project <{1}> already exists in space <{0}>"`

**Comment by tomasdavidorg** [comment]
Jenkins execute full downstream build

File: kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/pom.xml (hunk 1 above)
Comment (adrielparedes) [reply kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/pom.xml]: @paulovmr In some tests, if I don't add this I can't see the logs while tests are executed. That is why I added the test scope, to be sure is not added into final build

**Comment by mbiarnes** [comment]
@adrielparedes could you fix these conlficts - so we can merge?

**Comment by adrielparedes** [comment]
@mbiarnes done!

**Comment by mbiarnes** [comment]
Jenkins execute full downstream build

**Comment by mbiarnes** [comment]
@adrielparedes seems to fail

**Comment by mbiarnes** [comment]
@adrielparedes one test failing: https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/pullrequest/job/kie-wb-common-pullrequests/985/testReport/

**Comment by mareknovotny** [comment]
@mbiarnes that failed test seems as some glitch in the filesystem? test resource is missing.

**Comment by mbiarnes** [comment]
Jenkins execute full downstream build

**Comment by adrielparedes** [comment]
@mbiarnes @mareknovotny I've just run this on my local and everything runs OK, did something changed in FS/Jenkins?

**Comment by paulovmr** [comment]
Jenkins execute full downstream build

**Comment by paulovmr** [comment]
Jenkins execute full downstream build

**Comment by mareknovotny** [comment]
@adrielparedes the failure doesn't relates to the Jenkins as the changed test method existSpace is here in the PR changed, are you sure that you ran it locally with maven clean and the test resource for git repository wasn't there created/initialized before?
It seems to me that path 
`Caused by: java.nio.file.NoSuchFileException: src/test/resources/niogit-test`
really doesn't exist and failure in test `org.kie.workbench.common.screens.examples.backend.server.ExamplesServiceImplTest.testExistSpace` is valid. I can reproduce it on my local checkout of this PR too.

```
org.kie.workbench.common.screens.examples.backend.server.ImportExamplesException: Can't read spaces directory
	at org.kie.workbench.common.screens.examples.backend.server.ExamplesServiceImpl.existSpace(ExamplesServiceImpl.java:448)
	at org.kie.workbench.common.screens.examples.backend.server.ExamplesServiceImplTest.testExistSpace(ExamplesServiceImplTest.java:575)
```

**Comment by adrielparedes** [comment]
@mareknovotny I've just run them again on my local just with maven for me are green... and actually those tests were green last week, that is what is weird for me

**Pull request merged**